core/dbt/artifacts/schemas/manifest/v12/manifest.py

```diff
@@ -163,6 +163,10 @@ class WritableManifest(ArtifactMixin):
             description="The unit tests defined in the project",
         )
     )
+    functions: Mapping[UniqueID, Function] = field(
+        default_factory=dict,
```

Comment:
I'm not sure if this is actually how we want to do this. However, without doing this, the `WritableManifest` will break on loading of older manifests that don't have `functions`. The alternative to this would be to bump the schema version (v12 -> v13) and create an upgrade in `upgrade_manifest.py`.